src/pages/repository_settings_secret/content.js

```diff
@@ -0,0 +1,154 @@
+  }
+
+  render() {
+
+    const { owner, name } = this.props.params;
+    const { secrets } = this.props;
+    if (!secrets) {
+      return (
+        <div>Loading...</div>
+      );
+    }
+
+    return (
+      <PageContent className="repository-secrets">
+        {
+          secrets.map((secret, index) => {
+            return (
+
+              <div key={index}>
+                <Textfield
+                  label="Secret Name"
+                  floatingLabel
+                  style={{ width: '300px' }}
+                  value={secret.name}
+                  disabled={true}
+                />
+                <Textfield
+                  label="Update Secret Value"
+                  floatingLabel
+                  style={{ width: '300px' }}
+                  value={this.state[secret.name]}
+                  onChange={(field) => this.handleSecretChange(secret.name, field.target.value)}
+                />
+                <FABButton ripple mini
+                           onClick={this.handleDelete.bind(this, owner, name, secret.name)}
+                >
+                  <Icon name="delete"/>
+                </FABButton>
+                <FABButton ripple mini disabled={!this.state[secret.name]}
+                           onClick={this.handleUpdateSecret.bind(this, owner, name, secret.name)}
+                >
+                  <Icon name="update"/>
+                </FABButton>
+                <Switch id="push" disabled={true} checked={ secret.event.indexOf('push') !== -1}>Push</Switch>
+                <Switch id="tag" disabled={true} checked={ secret.event.indexOf('tag') !== -1}>Tag</Switch>
+                <Switch id="deployment" disabled={true}
+                        checked={ secret.event.indexOf('deployment') !== -1}>Deployment</Switch>
+                <Switch id="pull_request" disabled={true} checked={ secret.event.indexOf('pull_request') !== -1}>Pull
+                  request</Switch>
+                <hr/>
+              </div>
+            );
+          })
+        }
+
+
+        <Textfield
+          label="Secret Name"
+          floatingLabel
+          style={{ width: '300px' }}
+          onChange={(field) => this.handleAddKeyChange(field.target.value)}
+          value={this.state.addKey}
+        />
+        <Textfield
+          label="Secret Value"
+          floatingLabel
+          style={{ width: '300px' }}
+          onChange={(field) => this.handleAddSecretChange(field.target.value)}
+          value={this.state.addValue}
+        />
+        <FABButton ripple mini
+                   onClick={this.handleAdd.bind(this, owner, name)}
+        >
+          <Icon name="add"/>
+        </FABButton>
+        <hr/>
+
+      </PageContent>
+    );
+  }
+
+
+  handleDelete(owner, name, secret) {
+    events.emit(DEL_REPO_SECRET, { owner, name, secret });
+  }
+
+  handleUpdateSecret(owner, name, secretKey) {
+    //TODO: need to handle skip_verify and what events
```

Comment:
right now I'm just posting again the secret if a value has been typed in and it updates the value.  The events can't be updated since it would clear out the value but I still think it's worth as least updating the value?  Let me know what you think